New v1.1 navigation menu

RichieEscarez opened the pull request:
The nav menu includes the new and previously missing topics, see it in my fork: http://richieescarez.github.io/kubernetes/v1.1/index.html

In this PR I also have also:
- integrated this new nav menu into docwithnav.html template (so all files that we push into the _v1.1 folder in gh-pages will be build with this nav menu)
- included new v1.1 copies of the landing pages (which now all contain short descriptions (attempt to improve SEO))
- see specific details below

Please review the following changes to ensure they have been located correctly in the menu.

IMPORTANT: we have not run the scripts to generate the .html files nor have they been copied into the new _v1.1 folder in gh-pages. Therefore no topics will show when you click in the nav menu. 
~~**QUESTION** Can someone help run that script so that we can test the doc set out on k8s.io? _(it will publish the content but nobody will know it there unless they manually change their URLs for .../v1.0x/... to .../v1.1/...)_~~ thanks @krousey !

**Attn:**
@caesarxuchao, @nikhiljindal, @krousey, @thockin, @bprashanth, @jszczepkowski, @erictune 
Please see the location of your documents below and provide feedback on what needs to change (is anything missing? are the URLs correct?).

Notable changes:
- New "adv topics" landing page (please click link and review all, including short description):   http://richieescarez.github.io/kubernetes/v1.1/advancedtopics.html  
  ![image](https://cloud.githubusercontent.com/assets/11762685/10684480/81cc6238-7902-11e5-85a2-eaa7dc292331.png)
- The App. Admin. section was moved around a litte. I also removed from the menu and instead inserted mention of a "quick walkthrough" in the landing page of the "detailed walkthrough":
  - Review "detailed walkthrough" landing page (and new short descriptions): http://richieescarez.github.io/kubernetes/v1.1/app-admin-detailed.html
  - Moved: "Quick Start"         Inserted: "prereqs" "compute resources"
    ![image](https://cloud.githubusercontent.com/assets/11762685/10684591/cb86e4a6-7903-11e5-9677-1653d4519314.png)
    - Indented DNS and OVS:  
      ![image](https://cloud.githubusercontent.com/assets/11762685/10684713/71faaa56-7905-11e5-9491-bb89da51987b.png)
- Added "Extensions API" content: http://richieescarez.github.io/kubernetes/v1.1/api-ref.html
- Revised title: Added "installing" to our "Getting Started" topic title (SEO recommendation)  
  ![image](https://cloud.githubusercontent.com/assets/11762685/10697793/8c80a4ae-7962-11e5-99b9-63bc3f895681.png)
- Review these new short descriptions in the landing pages (SEO recommendation):  
  - http://richieescarez.github.io/kubernetes/v1.1/gs-localmachine.html
  - http://richieescarez.github.io/kubernetes/v1.1/gs-turnkey.html
  - http://richieescarez.github.io/kubernetes/v1.1/gs-custom.html
  - http://richieescarez.github.io/kubernetes/v1.1/basicstutorials.html
  - http://richieescarez.github.io/kubernetes/v1.1/deploy-clusters.html
  - http://richieescarez.github.io/kubernetes/v1.1/multi-tier.html
  - http://richieescarez.github.io/kubernetes/v1.1/setup-config.html
- Added/Inserted the following topics (none of these were perviously linked to the nav but other topics point to and open them (so they should appear in the nav to show context)):  
  - "life of pod" "firewalls" and "PV"  
    ![image](https://cloud.githubusercontent.com/assets/11762685/10684454/3903e85a-7902-11e5-94bc-654a7a91533a.png)  
  - all three sub topics  
    ![image](https://cloud.githubusercontent.com/assets/11762685/10684506/e9162370-7902-11e5-95c4-5a284c629473.png)  
  - "security context" "config best practices"  
    ![image](https://cloud.githubusercontent.com/assets/11762685/10684663/ce739640-7904-11e5-8b1f-d3783a6977d0.png)
  - "accessing the cluster" "service accounts" "etcd"  
    ![image](https://cloud.githubusercontent.com/assets/11762685/10684723/9bcb27ac-7905-11e5-8b12-ee573a902353.png)
  - "kubeconfig" "sharing cluster access" "Garbage collection"  
    ![image](https://cloud.githubusercontent.com/assets/11762685/10684761/03a4227a-7906-11e5-98ff-639d8cfa27e6.png)
  - Under Examples: "resource quotas" "setting CPU/mem limits"  
    ![image](https://cloud.githubusercontent.com/assets/11762685/10684770/21b7e792-7906-11e5-92f0-5d47b2e7b4b8.png)
  - "jsonpath" "all the kubectl commands" "kubectl for docker users" "release roadmap"  
    ![image](https://cloud.githubusercontent.com/assets/11762685/10684780/509d13fc-7906-11e5-9153-e97591f268d6.png)

Fixes #15903 
Related #15893

RichieEscarez committed:
New v1.1 navigation menu (with new and previously missing topics), integrate new doc set into docwithnav.html template, and included new copies of the landing pages (that contain short descriptions)
Fixes #15903

bgrant0607 commented:
We need to distinguish between menu selections that expand into more items in the menu bar and those that change the content on the RHS.

bgrant0607 commented:
I'm going to make a bunch of comments, but have a meeting shortly. If you have other things to do, you might want to wait for more feedback.

One thing discussed over IM: I don't like "Advanced Topics". These are really "New Features". Other issues with that menu and page:
* All the APIs are part of the "extensions" API group currently, including Ingress. I don't see why that is called out differently.
* Ingress, Job, and Horizontal Pod Autoscaling are all beta and available by default.
* DaemonSet and Deployment are disabled by default. Although they are reachable via /apis/extensions/v1beta1, they are NOT beta.

RichieEscarez commented:
Thanks @bgrant0607 , i will look into adding chevrons for container topics (that open sub-menus).

Sorry, I understood that the ingress resource was actually apart of core but ill move them all together.

The alpha items/topics should explicitly call out that they are alpha (as mentioned in discussions with dev owners). Should we try to call them out in the nav menu too? I was trying to avoid mentioning both alpha and beta all together.

bgrant0607 commented:
"Overview and concepts" should not be the first subgroup under "User's Guide". Ingress, Job, and Horizontal Pod Autoscaling should be added to that subgroup.

Also, "Architecture" doesn't belong under "Overview and concepts". It belongs under Cluster Administration.

bgrant0607 commented:
Put docs/user-guide/overview.html under Getting Started.

Put the concept list under Application Administration > Overview and concepts.

bgrant0607 commented:
Take Life of a Pod and Firewalls out of the concept list. Don't indent Persistent Volumes, as it looks out of place.

bgrant0607 commented:
Remove "Kubernetes Basics" from Tutorials.

Rename "Deploying Clusters" to "Deploying Clustered Applications". It's longer, but "Clusters" is ambiguous with Kubernetes clusters.

bgrant0607 commented:
Or maybe just "Clustered Applications"

bgrant0607 commented:
"Persistent Images" is incorrect terminology. It should be "Persistent Volumes".

bgrant0607 commented:
Put JSONPath under CLI Reference.

bgrant0607 commented:
The difference between "Overview" and "kubectl" is not clear. I suggest "Command Reference" for the latter.

bgrant0607 commented:
Done with my current round of comments.

RichieEscarez commented:
Ok thanks. Ill be offline for awhile and will recover these later today.

I just pushed these sub menu indicators (carets), PTAL:

![image](https://cloud.githubusercontent.com/assets/11762685/10702090/5c480916-797c-11e5-9369-e8690f2c9979.png)

bgrant0607 commented:
The carets help a lot, thanks.

RichieEscarez commented:
* Implemented all request/suggested changes (see list below for deviations/notes)

Other notable changes:
 * Fixed carets: moved to the left side (to avoid weird wrapping, see screen cap below)
 * Removed sub section: "Adv topics" (our "extensions" topics were all moved under app admin concepts)
 * Remove "life of a pod" and "firewalls" (neither are connected to the nav menu - user wont see their location when viewing those topics):  
    ```
    <li class="navSubMenu">  
	<a href="{{ site.url }}/v1.1/docs/user-guide/pod-states.html" class="nav-link">The Life of A Pod</a>  
    </li>  
    <li class="navSubMenu">  
	<a href="{{ site.url }}/v1.1/docs/user-guide/services-firewalls.html" class="nav-link">Firewalls</a>  
    </li>
    ```
 * Moved "k8s basics tutorials" into App. Admin. section: "Quick Walkthrough: Kubernetes Basics"
 * Added section/titles to app admin (to organize and improve scanability/retrievability):  
![image](https://cloud.githubusercontent.com/assets/11762685/10712202/018c49b6-7a48-11e5-8cd1-1984c1ade52c.png)

RichieEscarez commented on _layouts/docwithnav.html:
```diff
@@ -16,7 +16,12 @@
 		<!-- Navigation menu -->
 		<div class="col-md-3 hidden-sm hidden-xs" id="wrapper">
-			{% include nav_v1.0.html %}
+			{% case page.collection %}
```
this adds new _v1.1 content into the build and specifies what version of the nav menus to use per "collection" (ie _v1.0 or _v1.1)

RichieEscarez commented on _sass/_navigation.scss:
```diff
@@ -59,24 +58,13 @@ ul, ul ul {
-.menu > li > ul li:hover a,
-.menu > li > ul li:hover a:before {
-	color:#428BCB;
+/* add caret to container topics (that have sub-menus) */
```
new carets (container indicators for sub-menus)

RichieEscarez commented on _sass/_navigation.scss:
```diff
@@ -35,8 +35,7 @@ ul, ul ul {
 .menu > li > a,
-.menu > li > ul > li,
-.menu > li > ul > li > ul > li {
+.menu > li > ul > li{
```
CSS clean up (removed unused entries). Other uncommented changes in this file are also clean up.

RichieEscarez commented on _sass/_navigation.scss:
```diff
@@ -84,16 +72,16 @@ ul, ul ul {
 	font-weight:bold;
 	margin-top:2px;
-	margin-left:.6em;
+	margin-left:.3em;
```
shifting nav menu to accomodate carets (same with other margin-left changes below)

RichieEscarez commented:
@bgrant0607 PTAL

bgrant0607 commented:
MUCH better. Thanks. LGTM.

wojtek-t commented:
@RichieEscarez @k8s-oncall

There seem to be some problem with this PR - we cannot start any tests for it.

krousey commented:
@wojtek-t This branch has no tests.

brendandburns commented:
manually merging because merge bot is broken.
